### PR TITLE
Feature: Add new field types to the Fields API

### PR DESCRIPTION
**Pull request by pauloiankoski**
## Description

This PR adds a new field for the Fields API (`MultiSelect`) and add changes to Date and Phone fields to adding a new attribute to store the formatting string.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/Framework/FieldsAPI/Date.php [hunk 1]
```diff
@@ -15,4 +16,29 @@ class Date extends Field
     use Concerns\HasDescription;
 
     const TYPE = 'date';
+
+    /** @var string */
+    protected $dateFormat = '';
```
Comment (jonwaldstein) [inline src/Framework/FieldsAPI/Date.php:21]: @pauloiankoski do you think we should add a default format instead of an empty string?

Comment (pauloiankoski) [reply src/Framework/FieldsAPI/Date.php]: Added a default value reenforcing the same default value we have set for the template: `yyyy/mm/dd`

Comment (jonwaldstein) [reply src/Framework/FieldsAPI/Date.php]: Cool, that makes sense.  Should we do the same for `phoneFormat`?

Comment (pauloiankoski) [reply src/Framework/FieldsAPI/Date.php]: Nothing on `phoneFormat` is already an option. In the template, it'll renders a native `<input type="phone" />`.

**Review by jonwaldstein — APPROVED** [review]

**Pull request merged**